core/templates/hashfuncs.h

```diff
@@ -100,7 +100,160 @@ static inline uint32_t hash_one_uint64(const uint64_t p_int) {
 	return uint32_t(v);
 }
 
-static inline uint32_t hash_djb2_one_float(double p_in, uint32_t p_prev = 5381) {
+// Murmurhash2 32-bit version from https://github.com/aappleby/smhasher/blob/master/src/MurmurHash2.cpp.
```

Comment:
Seems all three thirdparty hashes you added are public domain, but this would be good to document properly with a quote of their license/public domain dedication, as the links can go 404.

Reply:
I replaced it with the sentence I found in the smhasher repo: "All MurmurHash versions are public domain software, and the author disclaims all copyright to their code."